projection/core/src/main/scala/com/lightbend/lagom/projection/State.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.lagom.projection
+
+import akka.annotation.ApiMayChange
+import com.lightbend.lagom.internal.projection.ProjectionRegistryActor.WorkerCoordinates
+
+import scala.util.control.NoStackTrace
+
+@ApiMayChange
+// TODO: provide serialisers (all the ADT is on LWWMaps so must be serializable)
+sealed trait Status
+sealed trait Stopped extends Status
```

Comment:
Why this sub-trait exists?

Reply:
This is an idiom I learnt from @renatocaval so we can implement the ADT using `case object` without paying the price of referring to the object type as `Stopped.type` all over the place and instead use `Stopped` as a type (which is the `trait` but the object satisfies equally).